int-set/src/lib.rs

```diff
@@ -374,11 +471,33 @@ impl Domain<u16> for u16 {
         true
     }
 
-    fn ordered_values() -> impl Iterator<Item = u32> {
+    fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
         (u16::MIN as u32)..=(u16::MAX as u32)
     }
 
-    fn ordered_values_range(range: RangeInclusive<u16>) -> impl Iterator<Item = u32> {
+    fn ordered_values_range(range: RangeInclusive<u16>) -> impl DoubleEndedIterator<Item = u32> {
```

Comment:
at some point I would probably write a macro do to these impls, since they're all so similar.

Reply:
Yeah, sounds good